### ISSUE TITLE
Debian 9 doesn't support rc.local

### ISSUE THREAD

**Issue by logzilla** [userify/shim#48]
Looks like Debian 9 is dropping support for rc.local.
I have no idea why people like breaking things that work, but it's the world we live in :)

![Why???](http://i0.kym-cdn.com/photos/images/original/000/097/668/WHY-WHY-WOULD-YOU-DO-THAT.jpg)
```
Creating Userify directory (/opt/userify/)
Creating uninstall script (/opt/userify/uninstall.sh)
Creating API login config (/opt/userify/creds.py)
Creating shim (/opt/userify/shim.{sh,py})
Removing exit 0 from rc.local (if there)
Checking Shim Startup
 
Unable to set start at bootup -- no /etc/rc.local file?
You'll have to set shim to startup on it's own: create an
init script that launches /opt/userify/shim.sh on startup.
In most distributions, this would have been a single line
in /etc/rc.local, but you may need to do something more
exotic. Please contact us with Linux version information
and we may have more information for you.
 
root@pve01 [~]: # cat /etc/os-release
PRETTY_NAME="Debian GNU/Linux 9 (stretch)"
NAME="Debian GNU/Linux"
VERSION_ID="9"
VERSION="9 (stretch)"
ID=debian
HOME_URL="https://www.debian.org/"
SUPPORT_URL="https://www.debian.org/support"
BUG_REPORT_URL=https://bugs.debian.org/
```

**Comment by jamiesonbecker** [comment]
Thanks mate. This is a top priority --  Debian is a fav of mine too. :)

**Comment by jamiesonbecker** [comment]
Was, anyway. Between dropping support for core filesystems running on remote servers (reiserfs), adopting systemd, etc, I'm on the lookout. Maybe I'll make my own..

**Comment by Jaxmetalmax** [comment]
Reading the installer script, the fedora aproach can be used, creating a systemd service would work

**Comment by jamiesonbecker** [comment]
@Jaxmetalmax That's a great idea. We'll do that, especially since systemd is widely supported now.

**Comment by duffytilleman** [comment]
Any update on this?  Looks like debian 9 is still not supported

**Comment by jamiesonbecker** [comment]
Hi Duffy, sorry for the delay -- announcement coming soon!

**Comment by jamiesonbecker** [comment]
Just as a heads-up for people who are waiting for this, Userify still works great on Debian 8 as well as Ubuntu 16.04 LTS, and you should be able to also do this on Debian 9, but you'll have to add this to the beginning of your install script:

```
touch /etc/rc.local
systemctl enable rc-local.service
```
as Jaxmetalmax mentioned above.

This is obviously not an awesome workaround, and we've got some great stuff coming VERY soon. Email beta@userify.com if you haven't already and are up for a bit of testing.

**Comment by jamiesonbecker** [comment]
(closed by accident.. that button's easy to hit!)

**Comment by userifydevteam** [comment]
This should be resolved by commit 251b6a6 and earlier commits and is now live on all Userify products. Please re-open if needed and thanks for using Userify!
